From 32c051ee1d76c33d389492b1cee23f0d1d752c55 Mon Sep 17 00:00:00 2001
From: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
Date: Fri, 30 Nov 2018 23:10:19 +0000
Subject: [PATCH] Move the "atspi2" target to a separate BUILD.gn file

//build/config/linux/BUILD.gn is read by //build/config/compiler/BUILD.gn,
which means it is processed by both host and target toolchains.

Consequently, having the "atspi2" target there means "gn gen" can fail
because the host lacks atspi-2's development files even though they are only
needed by the target toolchain during the build.

Bug: 879147
Change-Id: I8780e6e4d3a2b3a44fdb2cbbb0862a1a669acdaa
Reviewed-on: https://chromium-review.googlesource.com/c/1356979
Reviewed-by: Thomas Anderson <thomasanderson@chromium.org>
Reviewed-by: Dirk Pranke <dpranke@chromium.org>
Commit-Queue: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
Cr-Commit-Position: refs/heads/master@{#612831}
---
 build/config/linux/BUILD.gn        | 20 --------------------
 build/config/linux/atspi2/BUILD.gn | 29 +++++++++++++++++++++++++++++
 build/dotfile_settings.gni         |  1 +
 content/browser/BUILD.gn           |  2 +-
 content/test/BUILD.gn              |  2 +-
 5 files changed, 32 insertions(+), 22 deletions(-)
 create mode 100644 build/config/linux/atspi2/BUILD.gn

diff --git a/build/config/linux/BUILD.gn b/build/config/linux/BUILD.gn
index cd3e6eede1e9..6ca9b6a9348e 100644
--- a/build/config/linux/BUILD.gn
+++ b/build/config/linux/BUILD.gn
@@ -100,26 +100,6 @@ if (use_glib) {
   }
 }
 
-if (use_atk) {
-  pkg_config("atspi2") {
-    packages = [ "atspi-2" ]
-    atspi_version = exec_script(pkg_config_script,
-                                pkg_config_args + [
-                                      "atspi-2",
-                                      "--version-as-components",
-                                    ],
-                                "value")
-    atspi_major_version = atspi_version[0]
-    atspi_minor_version = atspi_version[1]
-    atspi_micro_version = atspi_version[2]
-    defines = [
-      "ATSPI_MAJOR_VERSION=$atspi_major_version",
-      "ATSPI_MINOR_VERSION=$atspi_minor_version",
-      "ATSPI_MICRO_VERSION=$atspi_micro_version",
-    ]
-  }
-}
-
 # Ensures all exported symbols are added to the dynamic symbol table.  This is
 # necessary to expose Chrome's custom operator new() and operator delete() (and
 # other memory-related symbols) to libraries.  Otherwise, they might
diff --git a/build/config/linux/atspi2/BUILD.gn b/build/config/linux/atspi2/BUILD.gn
new file mode 100644
index 000000000000..988a99568136
--- /dev/null
+++ b/build/config/linux/atspi2/BUILD.gn
@@ -0,0 +1,29 @@
+# Copyright 2018 The Chromium Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import("//build/config/linux/pkg_config.gni")
+import("//build/config/ui.gni")
+
+# These packages should _only_ be expected when building for a target.
+assert(current_toolchain == default_toolchain)
+
+if (use_atk) {
+  pkg_config("atspi2") {
+    packages = [ "atspi-2" ]
+    atspi_version = exec_script(pkg_config_script,
+                                pkg_config_args + [
+                                      "atspi-2",
+                                      "--version-as-components",
+                                    ],
+                                "value")
+    atspi_major_version = atspi_version[0]
+    atspi_minor_version = atspi_version[1]
+    atspi_micro_version = atspi_version[2]
+    defines = [
+      "ATSPI_MAJOR_VERSION=$atspi_major_version",
+      "ATSPI_MINOR_VERSION=$atspi_minor_version",
+      "ATSPI_MICRO_VERSION=$atspi_micro_version",
+    ]
+  }
+}
diff --git a/build/dotfile_settings.gni b/build/dotfile_settings.gni
index 4242ec695046..407a9cd14419 100644
--- a/build/dotfile_settings.gni
+++ b/build/dotfile_settings.gni
@@ -17,6 +17,7 @@ build_dotfile_settings = {
     "//build/config/linux/BUILD.gn",
     "//build/config/linux/pkg_config.gni",
     "//build/config/linux/atk/BUILD.gn",
+    "//build/config/linux/atspi2/BUILD.gn",
     "//build/config/linux/dri/BUILD.gn",
     "//build/config/mac/mac_sdk.gni",
     "//build/config/mac/rules.gni",
diff --git a/content/browser/BUILD.gn b/content/browser/BUILD.gn
index df5eef63cb12..64be02314118 100644
--- a/content/browser/BUILD.gn
+++ b/content/browser/BUILD.gn
@@ -2167,7 +2167,7 @@ jumbo_source_set("browser") {
 
     configs += [
       "//build/config/linux/atk",
-      "//build/config/linux:atspi2",
+      "//build/config/linux/atspi2",
     ]
 
     if (use_glib) {
diff --git a/content/test/BUILD.gn b/content/test/BUILD.gn
index b3eef03de313..a63a7c294712 100644
--- a/content/test/BUILD.gn
+++ b/content/test/BUILD.gn
@@ -431,7 +431,7 @@ jumbo_static_library("test_support") {
         [ "../browser/accessibility/accessibility_event_recorder_auralinux.cc" ]
     configs += [
       "//build/config/linux/atk",
-      "//build/config/linux:atspi2",
+      "//build/config/linux/atspi2",
     ]
   }
 
-- 
2.19.2

